File: src/main/java/com/jv/events/model/Event.java
package com.jv.events.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "events")
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String description;

    @Column(nullable = false)
    private LocalDateTime startDateTime;
    // separate date and time fields?
    @Column(nullable = false)
    private LocalDateTime endDateTime;
    // links local event to google calendar event so can update and delete events
    private String googleCalendarEventId;
    private String location;

}

File: frontend/app/src/main/java/com/northcoders/jv_events_platform_frontend/model/Event.kt
package com.northcoders.jv_events_platform_frontend.model

import java.time.LocalDateTime

data class Event(
    val id: Long = -1,  // Make id optional with default value
    val title: String,
    val description: String,
    val location: String,
    val startDateTime: String,
    val endDateTime: String,
    val googleCalendarEventId: String? = null
)